### PR TITLE
Implement bounding boxes for RandomCropAndResize

### PR DESCRIPTION
**Pull request by AmedeoBiolatti**
# What does this PR do?

Fixes # ([issue](https://github.com/keras-team/keras-cv/issues/733)) (typo in the issue, the class name is `RandomCropAndResize` not `RandomResizedCrop`)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

@divyashreepathihalli @LukeWood

### CODE REVIEW

**Comment by LukeWood** [comment]
@AmedeoBiolatti can you sign the CLA?  This Pr looks great, but we will need to have you sign the CLA to review and merge

**Comment by AmedeoBiolatti** [comment]
> @AmedeoBiolatti can you sign the CLA? This Pr looks great, but we will need to have you sign the CLA to review and merge

done!

**Comment by LukeWood** [comment]
> > @AmedeoBiolatti can you sign the CLA? This Pr looks great, but we will need to have you sign the CLA to review and merge
> 
> done!

Do you mind if I push some extremely minor updates to this?

**Comment by AmedeoBiolatti** [comment]
@LukeWood Do what's best for the project!

**Review by LukeWood — APPROVED** [review]

File: keras_cv/layers/preprocessing/random_crop_and_resize.py [hunk 1]
```diff
@@ -145,6 +148,62 @@ def augment_image(self, image, transformation, **kwargs):
...
+            [
+                (x1 - t_x1) / t_dx,
+                (y1 - t_y1) / t_dy,
+                (x2 - t_x1) / t_dx,
+                (y2 - t_y1) / t_dy,
+                rest,
+            ],
+            axis=-1,
+        )
+        output = tf.squeeze(output, axis=1)
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_crop_and_resize.py:168]: I think that you should be able to get rid of this squeeze by stacking on the correct axis.  If you need help debugging feel free to tf.print the shapes out and add them in code comments, I can try to help you structure it to fix this.

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py [hunk 2]
```diff
@@ -183,3 +186,91 @@ def test_augment_one_hot_segmentation_mask(self):
...
+            [
+                [[0, 0, 18, 18, 0], [8, 8, 18, 18, 0]],
+                [[8, 8, 18, 18, 0], [0, 0, 18, 18, 0]],
+            ]
+        )
+        expected_output = np.reshape(expected_output, (2, 2, 5))
+        self.assertAllClose(expected_output, output["bounding_boxes"])
+
+    def test_augment_bbox_ragged(self):
+        image = tf.zeros([2, 20, 20, 3])
```
Comment (bhack) [inline keras_cv/layers/preprocessing/random_crop_and_resize_test.py:249]: I think that we need to test also with ragged images with different resolutions or not?

File: keras_cv/layers/preprocessing/random_crop_and_resize.py (hunk 1 above)
Comment (AmedeoBiolatti) [reply keras_cv/layers/preprocessing/random_crop_and_resize.py]: Solved in a new commit (abb53b0d02a586f4eaf5be540f35c329cd0ff955)

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py (hunk 2 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/random_crop_and_resize_test.py:249]: Ragged inputs for crop and resize is essential.  If you can support it that would be great, but if you can’t get it to work we can merge and I can tackle ragged  inputs.

Comment (AmedeoBiolatti) [reply keras_cv/layers/preprocessing/random_crop_and_resize_test.py:249]: At the moment it seems that the class does not handle ragged images correctly, but maybe I miss something. Honestly it is a little outside the scope of this merge IMHO

Comment (LukeWood) [reply keras_cv/layers/preprocessing/random_crop_and_resize_test.py:249]: Ok, yeah let’s merge yours and I’ll add ragged support

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py [hunk 3]
```diff
@@ -183,3 +186,91 @@ def test_augment_one_hot_segmentation_mask(self):
...
+            [[[0, 0, 10, 10], [4, 4, 12, 12]], [[0, 0, 10, 10]]], dtype=tf.float32
+        )
+        bboxes = bounding_box.add_class_id(bboxes)
+        input = {"images": image, "bounding_boxes": bboxes}
+        mock_random = [0.0, 0.0, 0.1, 0.1]
+        layer = preprocessing.RandomCropAndResize(
+            target_size=(18, 18),
+            crop_area_factor=(0.5**2, 0.5**2),
+            aspect_ratio_factor=(1.0, 1.0),
+            bounding_box_format="xyxy",
```
Comment (bhack) [inline keras_cv/layers/preprocessing/random_crop_and_resize_test.py:260]: Do we have a preferred standard on the `bounding_box_format` in tests?

Comment (LukeWood) [reply keras_cv/layers/preprocessing/random_crop_and_resize_test.py:260]: thats a good question!  I think my preference would be to try to force a format conversion in unit tests, but no strong opinion.

**Comment by LukeWood** [comment]
Thank you to @AmedeoBiolatti for such a high quality contribution.  Only one review comment for such a good feature is epic!
